### ISSUE TITLE
Custom key of nuxt app context

### ISSUE THREAD

**Issue by chengzhuo5** [nuxt/nuxt#15234]
### Describe the feature

https://github.com/nuxt/framework/blob/02227a2f1b86b077082c50b46cb7486a8672eabe/packages/nuxt/src/app/nuxt.ts#L10
In micro frontend environment, multi apps will be conflicted with same context key (`nuxt-app`). Can make it configurable in nuxt.config.ts?

### Additional information

- [X] Would you be willing to help implement this feature?
- [X] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by odranoelBR** [comment]
Related to https://github.com/nuxt/nuxt/issues/18870

**Comment by victorgarciaesgi** [comment]
Is this issue still being discussed? For anyone searching for a tempory super hacky fix for micro-frontend, it can actually be solved with a rollup plugin

```ts
import replace from '@rollup/plugin-replace';

const replacePlugin = replace({
    delimiters: ['', ''],
    values: {
        __NUXT__: '__NUXT_MF__',
        'window.__NUXT__.config': 'window.__NUXT_MF__.config',
        [`getContext("nuxt-app")`]: `getContext("nuxt-app-mf")`,
    },
});


export default defineNuxtConfig({
    vite: {
        build: {
            rollupOptions: {
                plugins: [replacePlugin],
            },
        },
        plugins: [replacePlugin],
    },
    nitro: {
        rollupConfig: {
            plugins: [replacePlugin],
        },
    },
});

```

**Comment by nicolaspayot** [comment]
I believe this issue can be closed now as it was resolved with #27068.
